src/engine/server/server.cpp

```diff
@@ -1463,33 +1464,8 @@ int CServer::Run()
 
 			PumpNetwork();
 
-			if(ReportTime < time_get())
-			{
-				if(Config()->m_Debug)
-				{
-					/*
-					static NETSTATS prev_stats;
-					NETSTATS stats;
-					netserver_stats(net, &stats);
-
-					perf_next();
-
-					if(config.dbg_pref)
-						perf_dump(&rootscope);
-
-					dbg_msg("server", "send=%8d recv=%8d",
-						(stats.send_bytes - prev_stats.send_bytes)/reportinterval,
-						(stats.recv_bytes - prev_stats.recv_bytes)/reportinterval);
-
-					prev_stats = stats;
-					*/
-				}
-
-				ReportTime += time_freq()*ReportInterval;
-			}
-
-			// wait for incomming data
-			m_NetServer.Wait(5);
+			// wait for incoming data
+			m_NetServer.Wait(clamp(int((TickStartTime(m_CurrentGameTick+1)-time_get())*1000/time_freq()), 1, 1000/SERVER_TICK_SPEED));
```

Comment:
Don't think the higher wait time is worth it(increased cpu idle time) when the network traffic scales worse.

Reply:
Could half the maximum? Right now it's effectively 20 ms (the maximum time until the next tick is due). A maximum of 10 ms still gives 25% idle time improvement.

Reply:
Yeah, lets try that 👍 
There's probably lots of possible idle time improvement in the game server when a server is empty. But that's another issue.